app/queries/admin/gdpr_delete_requests_query.rb

```diff
@@ -0,0 +1,19 @@
+module Admin
+  class GDPRDeleteRequestsQuery
```

Comment:
Looks good @juliannatetreault, can we add some RSpec tests for this too please?

Reply:
Some thoughts: I suspect in the future we may be able to abstract out this class and `UsersQuery` class to be a `SearchQuery` class and use it across the admin in a more general purpose, but let's do that optimization when we find ourselves repeating the code too many times.

Reply:
Done! I also added issues for anything that can be addressed later that came up in the PR. 😊